common/src/primitives/id/mod.rs

```diff
@@ -143,6 +145,10 @@ impl<T> Id<T> {
         self.id
     }
 
+    pub const fn get_ref(&self) -> &H256 {
+        &self.id
+    }
```

Comment:
I don't think it's a conversion, it's rather an accessor for the lower-level representation of the id.

Also, above it there is the function
```
pub const fn get(&self) -> H256 {
    self.id
}
```
So, in `get_ref` the `get` part is more like a stem of the name rather than a prefix, and `ref` is the suffix.

IMO both names should be consistent with each other. So if the current variant feels off, I'd suggest one of the following:
1) Rename `get` to `inner` and `get_ref` to `inner_ref`.
2) Rename `get` to `block_hash` and `get_ref` to `block_hash_ref` (it's kind of lengthy, but better reflects the meaning of the value).
2) Only leave one function that returns a reference, name it either `get`, `inner` or `block_hash`.

Reply:
I think `fn to_hash(&self) -> H256` and `fn as_hash(&self) -> &H256` should work fairly well and agree with the API guidelines document linked above. Equivalent signatures named `to_inner`, `as_inner` also work. However, renaming the existing `get` method may involve updating a non-trivial number of call sites.

Reply:
> However, renaming the existing get method may involve updating a non-trivial number of call sites.

"Rename symbol" in VSCode does all the updates automatically, so it is trivial.
Ok, I've renamed them to `to_hash` and `as_hash` respectively and also renamed the `id: H256` field of `Id` to `hash` for consistency.